stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/view/MessageListView.kt

```diff
@@ -589,20 +596,31 @@ public class MessageListView : ConstraintLayout {
         buffer.enqueueData(listItem)
     }
 
-    private fun handleNewWrapper(listItem: MessageListItemWrapper) {
-        buffer.hold()
-
-        val isThreadStart = !adapter.isThread && listItem.isThread
-        val isOldListEmpty = adapter.currentList.isEmpty()
+    public fun setMessageListItemFilter(messageListItemFilter: MessageListItemFilter) {
+        check(::adapter.isInitialized.not()) { "Adapter was already initialized, please set MessageListItemFilter first" }
+        this.messageListItemFilter = messageListItemFilter
+    }
 
-        adapter.isThread = listItem.isThread
-        adapter.submitList(listItem.items) {
-            scrollHelper.onMessageListChanged(
-                isThreadStart = isThreadStart,
-                hasNewMessages = listItem.hasNewMessages,
-                isInitialList = isOldListEmpty && listItem.items.isNotEmpty()
-            )
-            buffer.active()
+    private fun handleNewWrapper(listItem: MessageListItemWrapper) {
+        CoroutineScope(DispatcherProvider.IO).launch {
+            messageListItemFilter.filterMessageListItem(listItem.items).let { filteredList ->
+                withContext(DispatcherProvider.Main) {
+                    buffer.hold()
+
+                    val isThreadStart = !adapter.isThread && listItem.isThread
+                    val isOldListEmpty = adapter.currentList.isEmpty()
+
+                    adapter.isThread = listItem.isThread
+                    adapter.submitList(filteredList) {
+                        scrollHelper.onMessageListChanged(
+                            isThreadStart = isThreadStart,
+                            hasNewMessages = listItem.hasNewMessages,
+                            isInitialList = isOldListEmpty && listItem.items.isNotEmpty()
```

Comment:
We're still using `listItem.items` instead of the filtered list here - is that intentional?

Reply:
Unintentional) Thanks for the catch!